posthog/hogql/test/test_parser_python.py

```diff
@@ -1,5 +1,414 @@
 from ._test_parser import parser_test_factory
+from posthog.hogql.ast import (
+    VariableAssignment,
+    Constant,
+    ArithmeticOperation,
+    Field,
+    ExprStatement,
+    Call,
+    ArithmeticOperationOp,
+    CompareOperationOp,
+    CompareOperation,
+    JoinExpr,
+    SelectQuery,
+    Program,
+    IfStatement,
+    Block,
+    WhileStatement,
+    Function,
+    Array,
+    Dict,
+)
+
+from posthog.hogql.parser import parse_program
+from posthog.hogql import ast
 
 
 class TestParserPython(parser_test_factory("python")):
-    pass
+    def _program(self, program: str, placeholders: dict[str, ast.Expr] | None = None) -> ast.Program:
+        return parse_program(program, placeholders=placeholders, start=None)
+
+    def test_program_variable_declarations(self):
```

Comment:
Let's not make `TestParserPython` diverge from `TestPythonCpp`. They should inherit all the same stuff from `parser_test_factory()`, so that the implementations stay in sync. Ofc we can move faster with the Python one, but once in master, we should converge ASAP @mariusandra

Reply:
Absolutely. Adding C++ parsing support for Hog and fixing this is somewhere high on my list.